RockWeb/Themes/NewSpring/assets/Lava/DATASETS/content-channel-items-by-channel.lava

```diff
@@ -54,95 +54,18 @@
-          {%- sql return:'parents' -%}
-            SELECT
-              ROW_NUMBER() OVER (ORDER BY ccia.[Order]) 'Index',
-              cci.Id,
-              cci.ContentChannelId 'ChannelId',
-              cci.StartDateTime 'PublishDateTime',
-              cci.ExpireDateTime 'ExpireDateTime',
-              ccis.Slug,
-              pd.AccessKey 'Dataset'
-            FROM ContentChannelItemAssociation ccia
-            JOIN ContentChannelItem cci
-            ON ccia.ContentChannelItemId = cci.Id
-            JOIN ContentChannelItemSlug ccis
-            ON ccis.ContentChannelItemId = cci.Id
-            JOIN AttributeValue av
-            ON av.EntityId = cci.ContentChannelId
-            JOIN PersistedDataset pd
-            ON pd.Id = av.Value
-            WHERE ccia.ChildContentChannelItemId = {{ item.Id }}
-            AND cci.ContentChannelId = {{ parentchannelid }}
-            AND av.AttributeId = 107096
-            ORDER BY ccia.[Order]
-          {%- endsql -%}
+          {% assign parentchannelid = parentchannelid | AsInteger %}
+          {% assign parents = 'content_parents' | PersistedDataset | Where:'Id', item.Id | Select:'Parents' | First | Where:'ChannelId', parentchannelid %}
         {%- endif -%}
-        {%- sql return:'related_items' -%}
-          SELECT TOP 9
-              ROW_NUMBER() OVER (ORDER BY COUNT(*) desc, cci2.ModifiedDateTime desc) 'Index',
-              cci2.Id,
-              cci2.ContentChannelId 'ChannelId',
-              cci2.StartDateTime 'PublishDateTime',
-              cci2.ExpireDateTime,
-              (
-                  SELECT Slug
-                  FROM ContentChannelItemSlug ccis
-                  WHERE ccis.ContentChannelItemId = cci2.Id
-              ) 'Slug',
-              (
-                  SELECT pd.AccessKey
-                  FROM AttributeValue av
-                  JOIN PersistedDataset pd
-                  ON av.Value = pd.Id
-                  WHERE av.EntityId = cci2.ContentChannelId
-                  AND av.AttributeId = 107096
-              ) 'Dataset'
-          FROM ContentChannelItem cci
-          JOIN TaggedItem ti
-          ON ti.EntityGuid = convert(nvarchar(50), cci.Guid)
-          JOIN TaggedItem ti2
-          ON ti.TagId = ti2.TagId
-          JOIN ContentChannelItem cci2
-          ON ti2.EntityGuid = convert(nvarchar(50), cci2.Guid)
-          WHERE cci.Id = {{ item.Id }}
-          AND cci2.Id != {{ item.Id }}
-          AND cci2.Status = 2
-          AND cci2.StartDateTime <= GETDATE()
-          AND (cci2.ExpireDateTime is NULL OR cci2.ExpireDateTime > GETDATE())
-          GROUP BY cci2.Id, cci2.ModifiedDateTime, cci2.StartDateTime, cci2.ExpireDateTime, cci2.ContentChannelId
-          ORDER BY COUNT(*) desc, cci2.ModifiedDateTime desc
-        {%- endsql -%}
+
+        {% assign related_items = 'content_related' | PersistedDataset | Where:'Id', item.Id | Select:'Related' | First %}
```

Comment:
Getting related items data from the related items dataset vs making another sql call. This dramatically speeds up build time.